### DISCUSSION TITLE
input type="hidden" doesnt works

### DISCUSSION THREAD

**Discussion by blinkinglight** [livewire/livewire#3678] [7. Bugs (archived)] [answered]
```blade
<div>
    <form method="post" action="" wire:submit.prevent="save">
        <input class="tagz" type="hidden" value="test" name="tags" wire:model="tags" />
        <input type="submit" value="submit" />
        <a href="javascript:;" onclick="$('.tagz').val('testrandomstrings');">test</a>
        {{ $tags }}
    </form>
</div>
```
or 
```blade
<div>
    <form method="post" action="" wire:submit.prevent="save">
        <div style="display:none;">
            <input class="tagz" type="text" value="test" name="tags" wire:model="tags" />
        </div>
        <input type="submit" value="submit" />
        <a href="javascript:;" onclick="$('.tagz').val('testrandomstrings');">test</a>
        {{ $tags }}
    </form>
</div>
```

with:

```php
<?php

namespace App\Http\Livewire;

use Livewire\Component;

class Publisher extends Component
{

    public $tags = "";

    public function save() {
        dd($this->tags);
    }

    public function render()
    {
        return view('livewire.publisher');
    }
}
```

problem: if field is hidden, php variables is not set. once i change type to text - everything works as expected;

is it known issue ? 

i am using this: https://selectize.dev/demos/2016/01/02/tagging/ ( in older version in makes input type="hidden" in newer = adds style display:none ) 

tested kind of everything. wire:model.lazy , wire:model.defer etc.

**Comment by joshhanley** [comment, marked as answer]
@blinkinglight Livewire doesn't pick up on changes to inputs if it's done by javascript. So instead you need to update the Livewire properties using javascript directly, which removes the need for a hidden input.

So using your example you would need to do something like this:

```blade
<div>
    <form method="post" action="" wire:submit.prevent="save">
        <input type="submit" value="submit" />
        <a href="javascript:;" onclick="@this.tags = 'testrandomstrings';">test</a>
        {{ $tags }}
    </form>
</div>
```

I have used the `@this` which is a Livewire helper for accessing properties from javascript, see here https://laravel-livewire.com/docs/2.x/inline-scripts

I've also removed the hidden input as it's no longer needed.

Hope this helps!

**Reply by blinkinglight** [reply]
> @blinkinglight Livewire doesn't pick up on changes to inputs if it's done by javascript. So instead you need to update the Livewire properties using javascript directly, which removes the need for a hidden input.
> 
> So using your example you would need to do something like this:
> 
> ```
> <div>
>     <form method="post" action="" wire:submit.prevent="save">
>         <input type="submit" value="submit" />
>         <a href="javascript:;" onclick="@this.tags = 'testrandomstrings';">test</a>
>         {{ $tags }}
>     </form>
> </div>
> ```
> 
> I have used the `@this` which is a Livewire helper for accessing properties from javascript, see here https://laravel-livewire.com/docs/2.x/inline-scripts
> 
> I've also removed the hidden input as it's no longer needed.
> 
> Hope this helps!

so... "Livewire doesn't pick up on changes to inputs if it's done by javascript." that means, no "tags" plugins, no WYSIWYG editors because they all uses javascript to update text fields?

**Reply by ryatkins** [reply]
Not necessarily.  You would use Alpine to integrate with them, listen for changes and send the data back and forth to Livewire via Alpine.  

I've you're a sponsor there is a video on integrating with Trix
https://laravel-livewire.com/screencasts/s4-rich-text

Also, see this Alpine integration page in the docs
https://laravel-livewire.com/docs/2.x/alpine-js

**Reply by blinkinglight** [reply]
ok thank you. i am sponsor, so i will watch that video ;)

**Reply by joshhanley** [reply]
@blinkinglight to expand on it some more, under the hood `wire:model` listens for an `input` event.

But when you change an input using javascript, that event doesn't get fired, where as it does when you manually change an input. 

So as Ryan said, you have to use something like Alpine to dispatch an input event when a programmatic change has occurred.

Definitely worth watching the videos 🙂

Hope this helps!

**Reply by blinkinglight** [reply]
i watched video already, and seems that to use alpine is not hard ;)
